### ISSUE TITLE
React Compiler with plugin-swc

### ISSUE THREAD

**Issue by samesfahani-tuplehealth** [vitejs/vite-plugin-react#428]
Will there be instructions on how to use the new React Compiler given that it's babel based?

**Comment by ArnaudBarre** [comment]
For now the simplest way is to switch back to the Babel plugin. They working a Rust port, I hope this can be builtin in SWC in the future!

**Comment by eeuneeun** [comment]
how can I New React Compiler setting Vite project, now?
I try to rewrite "vite.config.ts" file  (plus  plugins section) .....
but the "babel" letter makes errors

**Comment by alireza-akbarzadeh** [comment]
You should use vite-plugin-react

```ts
import { defineConfig } from "vite";
import react from "@vitejs/plugin-react";

const ReactCompilerConfig = {};

export default defineConfig({
  plugins: [
    react({
      babel: {
        plugins: [["babel-plugin-react-compiler", ReactCompilerConfig]],
      },
    }),
  ],
});
```

**Comment by matej-kaska** [comment]
Will be there an option to use react compiler with SWC in the future?

**Comment by ArnaudBarre** [comment]
If the React team lands the Rust port of the compiler, I think SWC will try to integrate it directly. But the rust part didn't get updated since annouced: https://github.com/facebook/react/commits/main/compiler/crates

**Comment by AaronDewes** [comment]
The Rust port has been removed: https://github.com/facebook/react/pull/32219

> For React Compiler though, we plan to continue using JavaScript and explore lightweight native wrappers for things like OXC and SWC plugins.

**Comment by ArnaudBarre** [comment]
Yeah, they still plan to have some integrations, but it will probably take times too

**Comment by rakleed** [comment]
See also https://github.com/oxc-project/oxc/issues/10048

**Comment by Hanaffi** [comment]
Any update on this please?

**Comment by HM-Suiji** [comment]
快一些，我很需要这个功能

**Comment by Arelav** [comment]
> 快一些，我很需要这个功能

它是开源的，如果紧急的话可以自己实现。
And we communicate her in English if you didn't notice

**Comment by HM-Suiji** [comment]
> > 快一些，我很需要这个功能
> 
> 它是开源的，如果紧急的话可以自己实现。 And we communicate her in English if you didn't notice

I am so sorry. I will try it.

**Comment by ianujvarshney** [comment]
I was also using `@vitejs/plugin-react-swc` and wanted React Compiler support.

After the Vite maintainer's clarification:
- No official way to use React Compiler without Babel yet
- `oxc-plugin-react-compiler` is unofficial (AI-generated, one-person project)
- Official approach: use `@vitejs/plugin-react` with Babel

I've migrated to `@vitejs/plugin-react` and accepted the Babel trade-off for now. Hoping the official Rust port from [PR #36173](https://github.com/facebook/react/pull/36173) lands soon.

Has anyone tested the performance difference between SWC and the new OXC-based plugin-react with React Compiler enabled?

**Comment by pippinmole** [comment]
The React Compiler port to Rust has been [merged into React](https://github.com/react/react/pull/36173) - when can we see this in the official Vite react plugin, if at all?

**Comment by ArnaudBarre** [comment]
Yeah I will see how we can integrate it in the Vite plugin soon!

**Comment by pippinmole** [comment]
> Yeah I will see how we can integrate it in the Vite plugin soon!

do you have an ETA? :)

**Comment by SimenB** [comment]
we have

```ts
  react({
    useAtYourOwnRisk_mutateSwcOptions(options) {
      options.jsc ??= {};
      options.jsc.transform ??= {};
      options.jsc.transform.reactCompiler = true;
    },
  }),
```

which works fine

**Comment by AndreFavotto** [comment]
+1 for this!

**Comment by kevinwt95** [comment]
Any updates here?
Is options.jsc.transform.reactCompiler = true a valid option? I can't find any documentation for it.

**Comment by orta** [comment]
v6.1.0 of this project now includes [experimental React Compiler support](https://github.com/vitejs/vite-plugin-react/releases/tag/plugin-react%406.1.0).

**Comment by AndreFavotto** [comment]
@orta The release mentions `@vitejs/plugin-react` only, not react-swc, unfortunately :/ 
Regardless, it is a good update. Considering switching to babel in the meantime.
